### PR TITLE
feat(scripts): allow custom interpreters

### LINKED ISSUE #907: Custom shell for scripts
I like writing my scripts in nushell, but there seems to be no way for `scripts` to emit custom shebang.

I mean, I could write my script exec like this: `"${pkgs.nushell}/bin/nu ./scripts/script-name.nu $@";`, but then I can't easily use custom packages without exposing them to the shell as well. It'd be great to be able to set `scripts.*.shell` or similar.

### PR DESCRIPTION
**Pull request by rawkode**
Closes #907

### CODE REVIEW

File: src/modules/scripts.nix [hunk 1]
```diff
@@ -1,22 +1,49 @@
...
-        default = "";
+  scriptType = types.submodule (
+    { config, ... }:
+    {
+      options = {
+        exec = lib.mkOption {
+          type = types.str;
+          description = "Shell code to execute when the script is run.";
+        };
+        interpreter = lib.mkOption {
```
Comment (domenkozar) [inline src/modules/scripts.nix:18]: I feel like we should call it `package` since we're calling it that way everywhere else too.

Comment (rawkode) [reply src/modules/scripts.nix:18]: No problem. Are you happy with binary, or is there a convention there too? Bin?

**Comment by domenkozar** [comment]
binary sounds good!

File: src/modules/scripts.nix (hunk 1 above)
Comment (rawkode) [reply src/modules/scripts.nix]: Fixed

File: src/modules/scripts.nix [hunk 2]
```diff
@@ -1,22 +1,49 @@
...
+    {
+      options = {
+        exec = lib.mkOption {
+          type = types.str;
+          description = "Shell code to execute when the script is run.";
+        };
+        package = lib.mkOption {
+          type = types.package;
+          description = "The package to use to run the script.";
+          default = "";
```
Comment (domenkozar) [inline src/modules/scripts.nix:21]:
```suggestion
          default = pkgs.bash;
```

**Review by domenkozar — CHANGES_REQUESTED** [review]

**Comment by rawkode** [comment]
Oops. I'm using the wrong formatter I think

**Comment by domenkozar** [comment]
@rawkode seems like you need to run `pre-commit run -a`

**Comment by rawkode** [comment]
@domenkozar I'm learning. Should be sorted now. Thanks for the assist.

**Comment by domenkozar** [comment]
No worries, that's why I'm here to help :)

File: src/modules/scripts.nix [hunk 3]
```diff
@@ -1,22 +1,48 @@
...
+        };
+        binary = lib.mkOption {
+          type = types.str;
+          description = "Override the binary name if it doesn't match package name";
+          default = config.package.pname;
+        };
+        description = lib.mkOption {
+          type = types.str;
+          description = "Description of the script.";
+          default = pkgs.bash;
```
Comment (domenkozar) [inline src/modules/scripts.nix:30]:
```suggestion
          default = "";
```

**Pull request merged**